### DISCUSSION TITLE
How to use defaultValues properly?

### DISCUSSION THREAD

**Discussion by AnthonyLzq** [airjp73/rvf#238] [Q&A] [answered]
I have a form that looks something like:

```js
<ValidatedForm
  validator={validator}
  method='post'
  resetAfterSubmit={true}
  id={FORM_UPDATE_ID}
  className={classes.form}
  defaultValues={{
    code: updateMemberResult.member.code,
    name: updateMemberResult.member.name,
    // ... other values
  }}
  onSubmit={(data, event) => {
    console.log('data, event', data, event)
  }}
>
  <FormInput
    name='name'
    inputClassName={cardClasses.cardFormControl}
    placeholder={`NOMBRES: ${updateMemberResult.member.name}`}
    marginBottom={32}
    marginBottomWithError={8}
  />
  {/* other fields */}
  <button type='submit'>Update</button>
</ValidatedForm>
```

But whenever I click the button my action is not being called, in fact, nothing is happening, the log I put in the `onSubmit` function is not working and there is no error message either. For the reference, I'm using Material-UI, but I don't know if that is relevant. 

This is only happening when I use the `defaultValues` prop. Besides, I just verified that the button is triggering his own `onClick` function.

**Comment by airjp73** [comment]
Hi! Could you provide a github or codesandbox link with a reproduction? There are a lot of possible causes for the issue and I can't really tell just from the code you've provided here.

**Reply by AnthonyLzq** [reply]
Let me create a repository with the issue.

**Reply by AnthonyLzq** [reply]
Here you can find the [repo](https://github.com/AnthonyLzq/remix-validated-form-with-material). Just clone it, install its dependencies and try to search (using ID: 1) and then update the user.

**Reply by AnthonyLzq** [reply]
Here I created a [StackBlitz URL](https://stackblitz.com/edit/github-8s9uxm), for some reason CodeSandbox doesn't let me import a project from GH.

**Reply by AnthonyLzq** [reply]
Any suggestions?

**Reply by AnthonyLzq** [reply]
Hi @airjp73, could please help me?

**Reply by airjp73** [reply, marked as answer]
Hi! Sorry I missed this.

In your reproduction, the validator has a required field (`photo`) that isn't present in the form. So the submission is failing because the form is invalid, but the errors doesn't appear because there's no photo field.

**Reply by AnthonyLzq** [reply]
You saved me! Didn't notice that, thank you so much!
